jmx-scraper/src/main/java/io/opentelemetry/contrib/jmxscraper/client/JmxRemoteClient.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.jmxscraper.client;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.security.Provider;
+import java.security.Security;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.management.remote.JMXConnector;
+import javax.management.remote.JMXConnectorFactory;
+import javax.management.remote.JMXServiceURL;
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
+import javax.security.sasl.RealmCallback;
+
+public class JmxRemoteClient {
```

Comment:
nit: Since this class serves as a builder-pattern style object whose purpose is to create a `JMXConnector` at some point, we could call it `JMXConnectorBuilder`. It's further complicated by the fact that the thing you get from calling `connect()` still has `connect()` on it. :) Also the existence of `JMXConnectorFactory` makes it worse. Heh, these 20+ year old names that we're stuck with....

Anyway, I think `JMXConnectorBuilder` is slightly more agreeable.

Reply:
I agree, I've renamed it to `JmxConnectorBuilder` and renamed `connect()` to `build()` even if it actually establishes the connection.